src/LanguageServer/Impl/Implementation/Server.WorkspaceSymbols.cs

```diff
@@ -16,56 +16,95 @@
         private static int _symbolHierarchyMaxSymbols = 1000;
 
         public override async Task<SymbolInformation[]> WorkspaceSymbols(WorkspaceSymbolParams @params, CancellationToken cancellationToken) {
-            await WaitForCompleteAnalysisAsync(cancellationToken);
+            var symbols = await SymbolIndex.WorkspaceSymbolsAsync(@params.query, cancellationToken);
+            return symbols.Take(_symbolHierarchyMaxSymbols).Select(s => new SymbolInformation {
+                name = s.Name,
+                kind = (SymbolKind)s.Kind,
+                location = new Location {
+                    range = s.Range,
+                    uri = s.DocumentUri,
+                },
+                containerName = s.ContainerName,
+            }).ToArray();
 
-            var members = Enumerable.Empty<IMemberResult>();
-            var opts = GetMemberOptions.ExcludeBuiltins | GetMemberOptions.DeclaredOnly;
-
-            foreach (var entry in ProjectFiles) {
-                members = members.Concat(
-                    await GetModuleVariablesAsync(entry as ProjectEntry, opts, @params.query, 50, cancellationToken)
-                );
-            }
-
-            members = members.GroupBy(mr => mr.Name).Select(g => g.First());
-            return members.Select(ToSymbolInformation).ToArray();
+            // await WaitForCompleteAnalysisAsync(cancellationToken);
+            // 
+            // var members = Enumerable.Empty<IMemberResult>();
+            // var opts = GetMemberOptions.ExcludeBuiltins | GetMemberOptions.DeclaredOnly;
+            // 
+            // foreach (var entry in ProjectFiles) {
+            //     members = members.Concat(
+            //         await GetModuleVariablesAsync(entry as ProjectEntry, opts, @params.query, 50, cancellationToken)
+            //     );
+            // }
+            // 
+            // members = members.GroupBy(mr => mr.Name).Select(g => g.First());
+            // return members.Select(ToSymbolInformation).ToArray();
         }
 
         public override async Task<SymbolInformation[]> DocumentSymbol(DocumentSymbolParams @params, CancellationToken cancellationToken) {
-            await WaitForCompleteAnalysisAsync(cancellationToken);
-            var opts = GetMemberOptions.ExcludeBuiltins | GetMemberOptions.DeclaredOnly;
-            var entry = ProjectFiles.GetEntry(@params.textDocument.uri);
-
-            var members = await GetModuleVariablesAsync(entry as ProjectEntry, opts, string.Empty, 50, cancellationToken);
-            return members
-                .GroupBy(mr => mr.Name)
-                .Select(g => g.First())
-                .Select(ToSymbolInformation)
-                .ToArray();
+            // var uri = @params.textDocument.uri;
+            // var symbols = await SymbolIndex.HierarchicalDocumentSymbolsAsync(uri, cancellationToken);
+            // var flat = symbols.Flatten(uri);
+            
+            throw new Exception("this function is not called");
```

Comment:
`NotImplementedException`?

Reply:
This is all WIP, this is a temporary thing I did to "implement" the function. I have no idea if this is called from VS or not.

Reply:
I ported the VS-specific functionKind, so resolving this.